src/loggers.py

```diff
@@ -0,0 +1,83 @@
+from __future__ import print_function
+import matplotlib.pyplot as plt
+import IPython.utils.warn
+import logging
+import sys
+import traceback
+
+class BqlLogger(object):
+  '''A logger object for BQL.
+
+     The idea of having a custom one is to make it easy to adapt to other
+     loggers, like python's builtin one (see LoggingLogger below) or the one
+     for iPython notebooks (see IpyLogger below), or for testing, or to set
+     preferences (see DebugLogger, QuietLogger, SilentLogger below).
+
+     Loggers should implement functions with the signatures of this base class.
+     They are welcome to inherit from it.  Do not depend on return values from
+     any of these methods.
+  '''
+  def info(self, msg_format, *values):
+    '''For progress and other informative messages.'''
+    print(msg_format % values)
+  def warn(self, msg_format, *values):
+    '''For warnings or non-fatal errors.'''
+    print(msg_format % values, file=sys.stderr)
+  def plot(self, _suggested_name, _figure):
+    '''For plotting. Name is a string, figure a matplotlib object.'''
+    plt.show()
```

Comment:
plt.show will display all non-closed figures.  Is there a way to display just the one that was passed in?